pkg/skaffold/build/jib/jib_sync.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jib
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/filemon"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/util"
+	"github.com/pkg/errors"
+)
+
+var syncLists = make(map[string]SyncMap)
+
+type SyncMap struct {
+	Direct    []SyncEntry `json:"direct"`
+	Generated []SyncEntry `json:"generated"`
+}
+
+type SyncEntry struct {
+	Src  string `json:"src"`
+	Dest string `json:"dest"`
+
+	filetime time.Time
+}
+
+func InitSync(ctx context.Context, workspace string, artifact *latest.JibArtifact) error {
+	syncMap, err := getSyncMap(ctx, workspace, artifact)
+	if (err != nil) {
+		return err
+	}
+	syncLists[artifact.Project] = *syncMap
```

Comment:
Is it possible that multiple Java projects have the same "JibArtifact.Project" name as a key to the map? The map is a global map, isn't it?

And isn't `artifact.Project == ""` for a single-module project setup?